### PR TITLE
Updates and Rename

### PR DESCRIPTION
**Pull request by davidtaylorhq**

### CODE REVIEW

File: config/locales/client.en.yml [hunk 1]
```diff
@@ -2,6 +2,5 @@ en:
   js:
     login:
       microsoft_office365:
```
Comment (gschlager) [inline config/locales/client.en.yml:4]: Is there a reason for not renaming the `microsoft_office365` key?

File: db/migrate/20211202134547_migrate_office365_user_info.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    INSERT INTO user_associated_accounts (
+      provider_name,
+      provider_uid,
+      user_id,
+      info,
+      last_used,
+      created_at,
+      updated_at
+    ) SELECT
+      'microsoft_office365',
```
Comment (gschlager) [inline db/migrate/20211202134547_migrate_office365_user_info.rb:15]: `microsoft_office365`?

File: plugin.rb [hunk 3]
```diff
@@ -1,65 +1,35 @@
...
-enabled_site_setting :office365_enabled
+enabled_site_setting :microsoft_auth_enabled
 
-class Office365Authenticator < ::Auth::OAuth2Authenticator
-  PLUGIN_NAME = 'oauth-office365'
+register_svg_icon "fab-microsoft"
 
+class ::MicrosoftAuthenticator < ::Auth::ManagedAuthenticator
   def name
     'microsoft_office365'
```
Comment (gschlager) [inline plugin.rb:17]: Why is the authenticator still called `microsoft_office365`?

File: config/locales/client.en.yml (hunk 1 above)
Comment (davidtaylorhq) [reply config/locales/client.en.yml:4]: Yes, this key is tied to the 'callback url'. If we change it, everyone would have to update their config with microsoft. I don't think it's worth that pain.

File: db/migrate/20211202134547_migrate_office365_user_info.rb (hunk 2 above)
Comment (davidtaylorhq) [reply db/migrate/20211202134547_migrate_office365_user_info.rb:15]: As above, we can't change it without a lot of pain

File: plugin.rb (hunk 3 above)
Comment (davidtaylorhq) [reply plugin.rb:17]: As above, we can't change it without a lot of pain

**Review by gschlager — APPROVED** [review]

**Review by pmusaraj — APPROVED** [review]

**Pull request merged**